core/include/mmcore/utility/platform/RuntimeInfo.h

```diff
@@ -15,25 +16,60 @@ class RuntimeInfo {
 public:
     static std::string GetHardwareInfo() {
         if (m_hardware_info.empty()) {
+            std::lock_guard<std::mutex> lock(write_mtx_);
             get_hardware_info();
         }
         return m_hardware_info;
     }
 
     static std::string GetOsInfo() {
         if (m_os_info.empty()) {
+            std::lock_guard<std::mutex> lock(write_mtx_);
             get_os_info();
         }
         return m_os_info;
     }
 
     static std::string GetRuntimeLibraries() {
         if (m_runtime_libraries.empty()) {
+            std::lock_guard<std::mutex> lock(write_mtx_);
             get_runtime_libraries();
         }
         return m_runtime_libraries;
     }
 
+    static std::string GetSMBIOSInfo() {
+        if (smbios_.empty()) {
+            std::lock_guard<std::mutex> lock(write_mtx_);
+            get_smbios_info();
+        }
+        return smbios_;
+    }
+
+    static std::string GetCPUInfo() {
+        if (cpu_.empty()) {
+            std::lock_guard<std::mutex> lock(write_mtx_);
+            get_cpu_info();
+        }
+        return cpu_;
+    }
+
+    static std::string GetGPUInfo() {
+        if (gpu_.empty()) {
+            std::lock_guard<std::mutex> lock(write_mtx_);
+            get_gpu_info();
+        }
+        return gpu_;
+    }
+
+    static std::string GetOSInfo() {
```

Comment:
What is the difference to `GetOsInfo`, do we really need this twice or can one be removed?